src/pool-bin/libraries/BinPool.sol

```diff
@@ -200,6 +201,82 @@ library BinPool {
         bytes32 feeForProtocol;
     }
 
+    function swapExactOut(State storage self, SwapParams memory params, uint128 amountOut)
+        internal
+        returns (BalanceDelta result, SwapState memory swapState)
+    {
+        Slot0 memory slot0Cache = self.slot0;
+        swapState.activeId = slot0Cache.activeId;
+        bool swapForY = params.swapForY;
+        swapState.protocolFee =
+            swapForY ? slot0Cache.protocolFee.getOneForZeroFee() : slot0Cache.protocolFee.getZeroForOneFee();
```

Comment:
Is the order reversed here ?

swapForY true => 0 -> 1 => getZeroForOneFee  but we've got `slot0Cache.protocolFee.getOneForZeroFee()`

Reply:
yeh true, copied from `swap()` -- this part is wrong -- will add test cases() around these later

Reply:
lolll, turned out was wrote by me